contracts/CryptoMarriage.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721BurnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+
+contract CryptoMarriage is
+  Initializable,
+  ERC721Upgradeable,
+  ERC721BurnableUpgradeable,
+  AccessControlUpgradeable
+{
+  using CountersUpgradeable for CountersUpgradeable.Counter;
```

Comment:
OZ's Counter is becoming a bit of a meme, you can replace it with a simple `uint`

Reply:
Good point ! 🙏